source/backend/api/Services/CompensationRequisitionService.cs

```diff
@@ -30,18 +30,28 @@ public PimsCompensationRequisition GetById(long compensationRequisitionId)
             return _compensationRequisitionRepository.GetById(compensationRequisitionId);
         }
 
-        public PimsCompensationRequisition Update(long compensationRequisitionId, PimsCompensationRequisition compensationRequisition)
+        public PimsCompensationRequisition Update(PimsCompensationRequisition compensationRequisition)
         {
             _user.ThrowIfNotAuthorized(Permissions.CompensationRequisitionEdit);
             compensationRequisition.ThrowIfNull(nameof(compensationRequisition));
 
-            if(compensationRequisitionId != compensationRequisition.CompensationRequisitionId)
+            _logger.LogInformation($"Updating Compensation Requisition with id ${compensationRequisition.CompensationRequisitionId}");
+
+            var currentCompensation = _compensationRequisitionRepository.GetById(compensationRequisition.CompensationRequisitionId);
+            var currentPayee = currentCompensation.PimsAcquisitionPayees.FirstOrDefault();
+            var updatedPayee = compensationRequisition.PimsAcquisitionPayees.FirstOrDefault();
+            if (currentPayee != null && updatedPayee != null)
             {
-                throw new BadRequestException("Invalid compensationRequisitionId.");
+                if (currentPayee.InterestHolderId == updatedPayee.InterestHolderId &&
```

Comment:
Hey, I don't really understand what the purpose of this is, am I being thick or does this need a comment?

Reply:
In a nutshell, the front end might override the id of the payee or completely remove it. However, that does not necessarily mean that it was updated. The code here makes sure that an update is actually needed by adding the missing payee id if necessary